autorest/azure/example/main.go

```diff
@@ -104,27 +150,92 @@ func getResourceGroups(client *autorest.Client) (*string, error) {
 		return nil, err
 	}
 
+	value := struct {
+		ResourceGroups []struct {
+			Name string `json:"name"`
+		} `json:"value"`
+	}{}
+
 	defer resp.Body.Close()
-	contents, err := ioutil.ReadAll(resp.Body)
+	dec := json.NewDecoder(resp.Body)
+	err = dec.Decode(&value)
 	if err != nil {
 		return nil, err
 	}
 
-	contentsString := string(contents)
+	var names = make([]string, len(value.ResourceGroups))
+	for i, name := range value.ResourceGroups {
+		names[i] = name.Name
+	}
+	return names, nil
+}
 
-	return &contentsString, nil
+func saveToken(spt azure.Token) {
```

Comment:
especially because in the callback I can't handle the error fine because this is all single-threaded code and the code is currently executing the request. so if I won't be able to save the token, the request might as well fail.